lib/screens/component.dart

```diff
@@ -0,0 +1,110 @@
+                  ChipComponentClass(
+                    color: Colors.grey,
+                    label: ChipText(text: 'Work in Progress'),
+                  ),
+                  ChipComponentClass(
+                    color: Colors.grey,
+                    label: Row(
+                      children: [
+                        ChipText(text: 'Work in Progress'),
+                        SpacerWidth(),
+                        Icon(
+                          Icons.more_horiz,
+                          color: Colors.black,
+                          size: 18,
+                        ),
+                      ],
+                    ),
+                  ),
+                  ChipComponentClass(
+                    color: Colors.white,
+                    label: Row(
+                      children: [
+                        ChipText(text: 'New'),
+                        SpacerWidth(),
+                        Icon(
+                          Icons.circle,
+                          color: Colors.red,
+                          size: 12,
+                        ),
+                      ],
+                    ),
+                  ),
+                  ChipComponentClass(
+                    color: Colors.white,
+                    label: ChipText(text: 'Old collection'),
+                  ),
+                ],
+              ),
+              //////////2
+              HeaderText(text: 'Circle icon'),
+              Row(
+                children: [
+                  CircleIconClass(
+                    icon: Icons.check_circle_outline_rounded,
+                    iconSize: 24,
+                    containerSize: 48,
+                    backgroundColor: Colors.greenAccent,
+                    iconColor: Colors.green,
+                  ),
+                  SpacerWidth(),
+                  CircleIconClass(
+                    icon: Icons.error,
+                    iconSize: 24,
+                    containerSize: 48,
+                    backgroundColor: Colors.orangeAccent,
+                    iconColor: Colors.orange,
+                  ),
+                  SpacerWidth(),
+                  CircleIconClass(
+                    icon: Icons.cancel,
+                    iconSize: 24,
+                    containerSize: 48,
+                    backgroundColor: Colors.redAccent,
+                    iconColor: Colors.red,
+                  ),
+                ],
+              ),
+              //////////3
+              HeaderText(text: 'Assignment Card'),
+              AssignmentCard(
+                title: 'Assignment V',
+                description:
+                    'Lorem ipsum dolor sit amet consectetur. Cum enim tempus purus condimentum tristique viverra. Ipsum aliquam nisl imperdiet justo arcu nulla tortor cursus suspendisse.',
+                status: ChipComponentClass(
+                  color: Colors.grey,
+                  label: ChipText(text: 'Work in Progress'),
+                ),
```

Comment:
This Chip component should be integrated into AssignmentCard it is part of a card

Reply:
But you should still pass text for chip through constractor